StatBoost.cpp

```diff
@@ -2743,6 +2738,125 @@ float CrewMember_Extend::CalculateStat(CrewStat stat, const CrewDefinition* def,
     return finalStat;
 }
 
+float CrewMember_Extend::CalculatePowerScaling(const StatBoost& statBoost)
+{
+    // Calculate the power scaling factor for a given ship
+    auto CalculatePowerShip = [&](int powerScalingShipId, bool mustBeCurrentRoom = false) -> float
+    {
+        // Calculate power scaling
+        int numPower = 0;
+        float sysPowerScaling = statBoost.def->powerScalingNoSys;
+        bool systemExists = false;
+        for (auto system : statBoost.def->systemPowerScaling)
+        {
+            if (system == SYS_ALL)
+            {
+                systemExists = true;
+                numPower += PowerManager::GetPowerManager(powerScalingShipId)->currentPower.second;
+            }
+            else if (system == SYS_REACTOR)
+            {
+                systemExists = true;
+                numPower += PowerManager::GetPowerManager(powerScalingShipId)->currentPower.first;
+            }
+            else
+            {
+                ShipManager* shipManager = G_->GetShipManager(powerScalingShipId);
+                if (shipManager != nullptr)
+                {
+                    if (shipManager->GetSystemRoom(system) != -1 && (!mustBeCurrentRoom || orig->iRoomId == shipManager->GetSystemRoom(system)))
```

Comment:
roomId isnt unique between player and enemy ships. To check if the crew is in the current room, the condition should be

```suggestion
orig->iShipId == powerScalingShipId && orig->iRoomId == shipManager->GetSystemRoom(system)
```

Reply:
Nvm I was stupid